### ISSUE TITLE
Fine-tune C3D with provided c3d_ucf101_finetune_whole_iter_20000. Different results.

### ISSUE THREAD

**Issue by gtln** [facebook/C3D#118]
Fine-tune C3D with provided c3d_ucf101_finetune_whole_iter_20000:
I adjusted the test_01.lst (paths to frames), run sh ucf101_testing.sh and I received:
... test_net.cpp:54] Test accuracy: 0.771599
According to your doc "This will give an accuracy of 80.19% (clip accuracy).".
The results are different. What can be the reason?

**Comment by dutran** [comment]
@gtln this may be due to the compression artifact when you extract frames. If you use ffmpeg to extract frames, you can may want to add '-q:v 1' into your command line to preserve the frame quality. I can share my extracted UCF101 frames with you shortly (subject to tarring & uploading time)

**Comment by dutran** [comment]
This is what I get on my side ... test_net.cpp:54] Test accuracy: 0.801862

**Comment by gtln** [comment]
Thank you, it works now, no need for sharing the UCF101 frames.
Is it possible to share c3d_ucf101_finetune_whole_iter_20000 for remaining 2 splits?

**Comment by dutran** [comment]
@gtln glad to hear that it works for you. What did you get? I do not have the fine-tuned models for the other 2 splits, but they are supposed to be easy for you to do that.

**Comment by gtln** [comment]
I got the same as you. Thank you.

**Comment by dutran** [comment]
@gtln could you tell @futurely how did you solve your issue?

**Comment by futurely** [comment]
The accuracy improved to 80.2% with the '-q:v 1' flag. It is still 2.1% lower than the C3D 1-net feature with linear SVM [1] and 2.4% (on UCF101 split 1) lower than the VGG-16 based spatial network of two-stream network fusion [2]. Using much deeper residual network [3] probably would improve the performance. BTW, TensorFlow just added support for [3d convolutions and pooling](https://github.com/tensorflow/tensorflow/commit/6a187ccddaebb741ea77fc3201c6e36625f0aadb) a few days ago.

[1] D. Tran, L. Bourdev, R. Fergus, L. Torresani, and M. Paluri, Learning Spatiotemporal Features with 3D Convolutional Networks, ICCV 2015.
[2] Christoph Feichtenhofer, Axel Pinz, Andrew Zisserman. Convolutional Two-Stream Network Fusion for Video Action Recognition. CVPR 2016.
[3] Kaiming He, Xiangyu Zhang, Shaoqing Ren, Jian Sun. Deep Residual Learning for Image Recognition. CVPR 2016.

**Comment by dutran** [comment]
@futurely there are two factor that makes this 80.2% it not comparable to the other numbers you provided: 1) This is clip-based accuracy and 2) this is on test split 1 only.

To make it more comparable, you need to:
1) fine-tune also on the other split.
2) use extract_image_features to extract 'prob' and aggregate them to make video-based accuracy.

**Comment by dutran** [comment]
@futurely I miss your "(on UCF101 split 1)". If that the case, then you only need to do the number 2).

**Comment by mangdian** [comment]
So if I want to fine-tune C3D in other datasets,such as ActivityNet and Thumos,I just need to adjust the trainlist file and testlist file ?
